sharding-core/src/main/java/io/shardingsphere/core/executor/sql/execute/result/AggregationDistinctQueryMetaData.java

```diff
@@ -0,0 +1,209 @@
+     * Get derived count column indexes.
+     *
+     * @return derived count column indexes
+     */
+    public Collection<Integer> getDerivedCountColumnIndexes() {
+        Collection<Integer> result = new LinkedList<>();
+        for (AggregationDistinctColumnMetaData each : columnMetaDataList) {
+            if (-1 != each.getDerivedCountIndex()) {
+                result.add(each.getDerivedCountIndex());
+            }
+        }
+        return result;
+    }
+    
+    /**
+     * Get derived sum column indexes.
+     *
+     * @return derived sum column indexes
+     */
+    public Collection<Integer> getDerivedSumColumnIndexes() {
+        Collection<Integer> result = new LinkedList<>();
+        for (AggregationDistinctColumnMetaData each : columnMetaDataList) {
+            if (-1 != each.getDerivedSumIndex()) {
+                result.add(each.getDerivedSumIndex());
+            }
+        }
+        return result;
+    }
+    
+    /**
+     * Get aggregation distinct column index.
+     *
+     * @param aggregationDistinctColumnLabel aggregation distinct column label
+     * @return aggregation distinct column index
+     */
+    public int getAggregationDistinctColumnIndex(final String aggregationDistinctColumnLabel) {
+        return Collections2.filter(columnMetaDataList, new Predicate<AggregationDistinctColumnMetaData>() {
+            @Override
+            public boolean apply(final AggregationDistinctColumnMetaData input) {
+                return aggregationDistinctColumnLabel.equals(input.getColumnLabel());
+            }
+        }).iterator().next().getColumnIndex();
+    }
+    
+    /**
+     * Get aggregation distinct column index.
+     *
+     * @param derivedSumIndex derived sum index
+     * @return aggregation distinct column index
+     */
+    public int getAggregationDistinctColumnIndex(final int derivedSumIndex) {
+        return Collections2.filter(columnMetaDataList, new Predicate<AggregationDistinctColumnMetaData>() {
+            @Override
+            public boolean apply(final AggregationDistinctColumnMetaData input) {
+                return derivedSumIndex == input.getDerivedSumIndex();
+            }
+        }).iterator().next().getColumnIndex();
+    }
+    
+    /**
+     * Get aggregation distinct column label.
+     *
+     * @param aggregationDistinctColumnIndex aggregation distinct column index
+     * @return aggregation distinct column label
+     */
+    public String getAggregationDistinctColumnLabel(final int aggregationDistinctColumnIndex) {
+        return Collections2.filter(columnMetaDataList, new Predicate<AggregationDistinctColumnMetaData>() {
+            @Override
+            public boolean apply(final AggregationDistinctColumnMetaData input) {
+                return aggregationDistinctColumnIndex == input.getColumnIndex();
+            }
+        }).iterator().next().getColumnLabel();
+        
+    }
+    
+    @RequiredArgsConstructor 
+    @Getter(AccessLevel.PRIVATE)
```

Comment:
Why need private getter method?

Reply:
Because i want those methods only to be used by its outer class.